webui/react/src/stores/users.tsx

```diff
@@ -98,68 +123,115 @@ export const useEnsureUsersFetched = (canceler: AbortController): (() => Promise
   if (context === null) {
     throw new Error('Attempted to use useUsers outside of Users Context');
   }
+  const config = cfg ?? {};
+  const usersPagination = context.usersByKey.get(encodeParams(config));
+
+  if (!usersPagination) return NotLoaded;
+
+  const userPage = {
+    pagination: usersPagination.pagination,
+    users: usersPagination.users.flatMap((userId) => {
+      const user = context.users.get(userId);
 
-  return context.users;
+      return user ? [user] : [];
+    }),
+  };
+
+  return Loaded(userPage);
 };
 
-export const useCurrentUsers = (): UseCurentUserReturn => {
+export const useUpdateCurrentUser = (): ((id: number) => void) => {
   const context = useContext(UsersContext);
 
   if (context === null) {
-    throw new Error('Attempted to use useCurrentUser outside of User Context');
+    throw new Error('Attempted to use useUpdateCurrentUser outside of Users Context');
   }
-  const { currentUser, updateCurrentUser, updateUsers } = context;
 
-  const userUpdateCallback = useCallback(
-    (user: DetailedUser, users: DetailedUser[] = []) => {
-      const usersArray = [...users];
+  const { updateCurrentUser } = context;
+  const callback = useCallback(
+    (id: number) => {
+      updateCurrentUser(() => Loaded(id));
+    },
+    [updateCurrentUser],
+  );
 
-      updateCurrentUser(() => {
-        const userIdx = usersArray.findIndex((changeUser) => changeUser.id === user.id);
+  return callback;
+};
 
-        if (userIdx > -1) usersArray[userIdx] = { ...usersArray[userIdx], ...user };
+export const useUpdateUser = (): ((
+  id: number,
+  updater: (arg0: DetailedUser) => DetailedUser,
+) => void) => {
+  const context = useContext(UsersContext);
 
-        return Loaded(user);
-      });
+  if (context === null) {
+    throw new Error('Attempted to use useUpdateUser outside of Users Context');
+  }
 
+  const { updateUsers } = context;
+  const callback = useCallback(
+    (id: number, updater: (arg0: DetailedUser) => DetailedUser) => {
       updateUsers((prevState) => {
-        if (!Loadable.isLoaded(prevState) || !usersArray.length) return prevState;
+        if (prevState.has(id)) {
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          return prevState.update(id, (detailedUser) => updater(detailedUser!));
+        }
 
-        if (isEqual(prevState.data, usersArray)) return prevState;
-
-        return Loaded(usersArray);
+        return prevState;
       });
     },
-    [updateCurrentUser, updateUsers],
+    [updateUsers],
   );
 
-  return {
-    currentUser,
-    updateCurrentUser: userUpdateCallback,
-  };
+  return callback;
+};
+
+export const useCurrentUser = (): Loadable<DetailedUser> => {
+  const context = useContext(UsersContext);
+
+  if (context === null) {
+    throw new Error('Attempted to use useCurrentUser outside of User Context');
+  }
+  const { currentUser, users } = context;
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
i think we should handle null

Reply:
dont we handle this?

Reply:
This is actually statically guaranteed to be non-null, hence the assertion

Reply:
Worthy of a comment in the code explaining that though.